### ISSUE TITLE
tsc & trailing-comma clash with spread operator

### ISSUE THREAD

**Issue by dsatfire** [palantir/tslint#3960]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: tslint-loader (webpack)

#### TypeScript code being linted

```ts
// code snippet
const {
    var1,
    var2,
    var3,
    var4,
    var5,
    var6,
    var7,
    ...restProps
} = this.props;
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:recommended",
    "rules": {
        "interface-name": [ true, "never-prefix"]
    }
}
```

#### Actual behavior
Rule tailling-comma is violated. 
```
Missing trailing comma
```
However, tsc will throw if comma is there.
```
TS1013: A rest parameter or binding pattern may not have a trailing comma.
```
#### Expected behavior
tslint recognizes the spread operator in use and does not mark rule as violated.
